### ISSUE TITLE
Idea for real offline messaging without servers

### ISSUE THREAD

**Issue by tox-user** [TokTok/c-toxcore#604]
Tox in order to be able to compete with many popular, less secure messaging apps needs offline messaging. There might be a way to achieve it without using servers.

Users could choose their friends to be trusted relays for their messages. Such friends would relay offline messages for them when they are offline and when their conversation target becomes online. Those relayed messages could either be readable by relays or not (additional encryption). Such messages can't be modified. Users need to be sure that their message will arrive untouched. Everyone would be able to choose if they want to be a relay for their friends or not. The more trusted friends we have the more reliable this kind of offline messaging would be. Offline messages should probably by text only or limited in size to not waste other people's bandwidth.

This is how it would work:
1. Alice wants to send a message to Bob, but he is offline. She sends him an offline message.
2. Alice goes offline.
3. Bob becomes online. If any of Alice's trusted friends are on, they will relay him the message from Alice. They received it earlier.
4. Bob can wait with his response until Alice becomes online or use his trusted friends in the same way to relay his message while he is offline.

**Comment by dvor** [comment]
Offline messages proposal https://github.com/anoadragon453/tox-offline-messaging-prop/pull/2

**Comment by SkyzohKey** [comment]
This issue has been opened 28 days ago and has not received any follow-back. Thus closing.

**Comment by robinlinden** [comment]
No, that's okay. Activity is slow right now, but hopefully we can talk about this soon.

**Comment by SkyzohKey** [comment]
@robinlinden Oh, ok sorry :)
